api-server/server/middlewares/csurf.js

```diff
@@ -7,8 +7,11 @@ export default function() {
     }
   });
   return function csrf(req, res, next) {
-    const path = req.path.split('/')[1];
-    if (/(^api$|^unauthenticated$|^internal$|^p$)/.test(path)) {
+    const pathArr = req.path.split('/');
+    if (/(^api$|^unauthenticated$|^internal$|^p$)/.test(pathArr[1])) {
+      return next();
+    }
+    if (/(^donate$)/.test(pathArr[1]) && /(^update-paypal$)/.test(pathArr[2])) {
```

Comment:
Why is this change required?

Reply:
Although csurf helps protect against cross-site request forgery attempts, I believe it also prevents us from processing legitimate webhook events. That is why I exempted the webhook route from CSRF protection.

That being said, we verify each events that hits that route and make sure they come from paypal before proceeding with modifying our records.

Reply:
We can unify the same into the previous check :)

Reply:
As discussed we do not need this anymore, but please add a comment block stating that any changes to this must accommodate for the fact that PayPal hits these without CSRF.